cpp/src/arrow/CMakeLists.txt

```diff
@@ -488,6 +488,9 @@ add_dependencies(arrow ${ARROW_LIBRARIES})
 
 if(ARROW_BUILD_STATIC AND WIN32)
   target_compile_definitions(arrow_static PUBLIC ARROW_STATIC)
+  if(ARROW_WITH_UTF8PROC)
+    target_compile_definitions(arrow_static PUBLIC UTF8PROC_STATIC)
```

Comment:
I think that it has a problem when we use shared utf8proc for the arrow static library.
How about using `INTERFACE_COMPILER_DEFINITIONS` property like https://github.com/apache/arrow/blob/master/cpp/cmake_modules/ThirdpartyToolchain.cmake#L2125-L2126 ?